src/symbolize/mod.rs

```diff
@@ -428,7 +431,7 @@ impl<R> Resolve for R where R: Symbolize + TranslateFileOffset {}
 ///
 /// Please refer to the [`Input`] enum for an overview of the various offset
 /// types.
-pub(crate) trait TranslateFileOffset
+pub trait TranslateFileOffset
```

Comment:
Unsure about this suggestion, to be honest. `Resolve` and `Symbolize` deal with symbols. In a module called `symbolize` it makes sense to me omit the "symbol" part from the name of the main traits to keep them concise. But `TranslateFileOffset` is at best remotely related to symbols, which is why I figured it would be better to be more verbose. If I see `Translate` in a symbolize module I'd suspect it's about symbol translation (e.g., one type into another or whatever; but the point is I wouldn't guess that it's about offset translation).
Re: future extension: To be honest, I'd rather introduce a separate trait. I believe it's more or less accepted practice that many traits are just comprised of a single method (`Deref`, `From`, `IntoIterator`, ...). That allows for more fine grained specification of requirements in call sites as well as the possibility to implement only what is truly necessary. Otherwise, if a type pops up that conceptually can only support one type of translation we may be in trouble, because we can reasonably only implement one of the methods.
I think we've seen a somewhat tangential problem play out with the original `SymResolver` trait. Check commit 7199fdc85bab6d8c203dd03861e0f199c5cdd70a:

> Ever since the beginning of time, most internals have evolved around the
> SymResolver trait. As we double down on providing inspection
> functionality (such as iteration over all symbols), this trait felt more
> a more like the wrong abstraction. The reason being that some symbol
> sources basically by design don't support such inspection. That led to
> shenanigans such as methods being effectively unimplemented, but
> returned an error, which then meant that we needed tests for this
> unimplemented functionality, to not have abysmal code coverage numbers.
> Long story short, the SymResolver trait as-is, is not quite the right
> abstraction.